src/java/nps/id/publicapi/java/client/AppListener.java

```diff
@@ -0,0 +1,149 @@
+package nps.id.publicapi.java.client;
+
+import com.nordpool.id.publicapi.v1.ConfigurationRow;
+import com.nordpool.id.publicapi.v1.ContractRow;
+import com.nordpool.id.publicapi.v1.order.*;
+import com.nordpool.id.publicapi.v1.order.request.OrderEntryRequest;
+import nps.id.publicapi.java.client.connection.clients.StompClient;
+import nps.id.publicapi.java.client.connection.clients.StompClientGenericFactory;
+import nps.id.publicapi.java.client.connection.enums.PublishingMode;
+import nps.id.publicapi.java.client.connection.enums.WebSocketClientTarget;
+import nps.id.publicapi.java.client.connection.storage.SimpleCacheStorage;
+import nps.id.publicapi.java.client.connection.subscriptions.exceptions.SubscriptionFailedException;
+import nps.id.publicapi.java.client.connection.subscriptions.requests.SubscriptionRequestBuilder;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.stereotype.Service;
+import nps.id.publicapi.java.client.security.options.CredentialsOptions;
+
+import java.time.ZonedDateTime;
+import java.util.Collections;
+import java.util.UUID;
+
+@Service
+public class AppListener implements ApplicationListener<ContextRefreshedEvent> {
+    private static final Logger logger = LogManager.getLogger(AppListener.class);
+
+    private final int _demoArea = 2;
+    private final String _clientId = UUID.randomUUID() + "-java-demo-client";
+
+    private final StompClientGenericFactory stompClientGenericFactory;
+
+    private final SubscriptionRequestBuilder subscribeRequestBuilder;
+
+    public AppListener(CredentialsOptions credentialsOptions, StompClientGenericFactory stompClientGenericFactory) {
+        this.stompClientGenericFactory = stompClientGenericFactory;
+        this.subscribeRequestBuilder = SubscriptionRequestBuilder.createBuilder(credentialsOptions.getUserName(), "v1");
+    }
+
+    @Override
+    public void onApplicationEvent(ContextRefreshedEvent event) {
+        try {
+            var edgeUiClient = CreateClient(WebSocketClientTarget.EDGE);
+            var middlewareUiClient = CreateClient(WebSocketClientTarget.MIDDLEWARE);
+
+            // Delivery areas
+            SubscribeDeliveryAreas(edgeUiClient);
+            SubscribeDeliveryAreas(middlewareUiClient);
+
+            // Contracts
+            SubscribeContracts(edgeUiClient, PublishingMode.CONFLATED);
+            SubscribeContracts(middlewareUiClient, PublishingMode.CONFLATED);
+
+            SubscribeContracts(edgeUiClient, PublishingMode.STREAMING);
+            SubscribeContracts(middlewareUiClient, PublishingMode.STREAMING);
```

Comment:
no need to consume streaming and conflated data at the same time as well

Reply:
Left only one streaming method subcription